backend/app/store/search/internal/elastic.go

```diff
@@ -0,0 +1,364 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/elastic/go-elasticsearch/v7/esutil"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/umputun/remark42/backend/app/store"
+	"github.com/umputun/remark42/backend/app/store/engine"
+	service "github.com/umputun/remark42/backend/app/store/search/service"
+)
+
+// elastic implements Service directly (not searchEngine)
+// because it has own mechanisms for batching
+// and do not require additional maintenance
+type elastic struct {
```

Comment:
This file is not tested at all, it shouldn't be like that.

Reply:
I've added some simple straight test, now coverage is about 40%. I've mocked elastic, but haven't mock meaningful response, it always returns 200 with empty body. But we test that requests are sent.